src/System.IO.Pipelines/StreamExtensions.cs

```diff
@@ -0,0 +1,244 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines
+{
+    public static class StreamExtensions
+    {
+        /// <summary>
+        /// Adapts a <see cref="Stream"/> into a <see cref="IPipelineWriter"/>.
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <returns></returns>
+        public static IPipelineWriter AsPipelineWriter(this Stream stream)
+        {
+            return (stream as IPipelineWriter) ?? stream.AsPipelineWriter(ArrayBufferPool.Instance);
+        }
+
+        /// <summary>
+        /// Adapts a <see cref="Stream"/> into a <see cref="IPipelineWriter"/>.
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <param name="pool"></param>
+        /// <returns></returns>
+        public static IPipelineWriter AsPipelineWriter(this Stream stream, IBufferPool pool)
+        {
+            var writer = new PipelineReaderWriter(pool);
+            writer.CopyToAsync(stream).ContinueWith((task) =>
+            {
+                if (task.IsFaulted)
+                {
+                    writer.CompleteReader(task.Exception);
+                }
+                else
+                {
+                    writer.CompleteReader();
+                }
+            });
```

Comment:
Ah yea, I was being lazy here. Though passing of those arguments is one of the reasons I try to avoid continue with...